### PR TITLE
[COST-3997] upgrade kombu

### PR DESCRIPTION
**Pull request by maskarb**
## Jira Ticket

[COST-3997](https://issues.redhat.com/browse/COST-3997)

## Description

This change will update kombu and some datetime/date cleanup.

## Testing

1. smokes

## Summary by Sourcery

Standardize date/time handling by centralizing parsing and conversion to UTC, migrate VMParams to Pydantic with validation, update SQL templates to consistently cast source_uuid as string, and bump kombu to the latest version.

New Features:
- Introduce to_utc_datetime util to normalize date/time inputs into UTC-aware datetime objects
- Upgrade kombu dependency to latest version in Pipfile

Enhancements:
- Refactor list_days and get_months_in_date_range to operate on date objects and use to_utc_datetime for parsing
- Migrate VMParams from a dataclass to a Pydantic BaseModel with integrated validation and UUID handling
- Standardize date usage across codebase and tests by consistently using date objects instead of zeroed datetimes
- Cast source_uuid as string in Trino SQL templates for AWS, Azure, and HCS queries
- Update settings.UTC definition for consistent timezone handling

Tests:
- Add comprehensive unit tests for to_utc_datetime covering various input formats and error cases
- Adjust existing tests to reflect new date object handling and updated utility behavior

## Summary by Sourcery

Standardize date handling and upgrade messaging dependency by centralizing date parsing, refactoring range utilities, migrating parameter classes to Pydantic, and updating SQL templates to correctly cast UUIDs

New Features:
- Introduce to_date utility to normalize string and datetime inputs into date objects

Enhancements:
- Refactor DateHelper.list_days, list_month_tuples, and get_months_in_date_range to operate on datetime.date and streamline range logic
- Migrate VMParams from dataclass to Pydantic BaseModel with start/end validation and UUID-typed source_uuid
- Cast source_uuid as string in Trino SQL templates for AWS, Azure, and HCS queries
- Update API endpoints and Celery tasks to use date objects instead of datetime strings

Build:
- Bump kombu dependency to latest version in Pipfile

Tests:
- Add unit tests for to_date and list_month_tuples covering diverse formats and edge cases
- Adjust existing tests to expect date objects across utilities and report code

### CODE REVIEW

File: koku/koku/settings.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ class CacheEnum(StrEnum):
 
 TIME_ZONE = "UTC"
 
-UTC = ZoneInfo("UTC")
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/koku/settings.py:351]: We use python 3.11.

File: koku/api/utils.py [hunk 2]
```diff
@@ -299,21 +302,19 @@ def list_days(self, start_date, end_date):
...
-        days = (end_midnight - start_midnight + self.one_day).days
+            start_date = start_date.date()
+        if isinstance(end_date, datetime.datetime):
+            end_date = end_date.date()
+        days = (end_date - start_date + self.one_day).days
 
         # built-in range(start, end, step) requires (start < end) == True
         day_range = range(days, 0) if days < 0 else range(days)
-        return [start_midnight + datetime.timedelta(i) for i in day_range]
+        return [start_date + datetime.timedelta(i) for i in day_range]
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/api/utils.py:316]: "range(days, 0) returns an empty range when days is negative" is false:
```
>>> for i in range(-10, 0):
...   print(i)
... 
-10
-9
-8
-7
-6
-5
-4
-3
-2
-1
>>> 
```

File: koku/api/utils.py [hunk 3]
```diff
@@ -494,18 +495,52 @@
...
+        return date.astimezone(tz=settings.UTC)
+
+    if isinstance(date, datetime.date):
+        # Convert date to datetime at midnight UTC
+        return datetime.datetime(date.year, date.month, date.day, tzinfo=settings.UTC)
+
+    raise TypeError(f"Expected str, datetime, date, or None, got {type(date)}")
+
+
 def get_months_in_date_range(
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/api/utils.py:547]: What is the code quality of `get_months_in_date_range` now?

File: koku/masu/processor/parquet/parquet_report_processor.py [hunk 4]
```diff
@@ -109,7 +109,7 @@ def account(self):
     @property
     def provider_uuid(self):
         """The provider UUID."""
-        return self._provider_uuid
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/masu/processor/parquet/parquet_report_processor.py]: This change was reverted.

File: koku/masu/database/report_manifest_db_accessor.py [hunk 5]
```diff
@@ -259,19 +259,19 @@ def set_manifest_daily_start_date(self, manifest_id, date):
...
+    def get_manifest_daily_start_date(self, manifest_id: int) -> date | None:
         """
         Get manifest processing daily archive start date.
         Used to prevent grabbing different starts from partial processed data
         """
         manifest = self.get_manifest_by_id(manifest_id)
         if manifest:
             if manifest.daily_archive_start_date:
-                return manifest.daily_archive_start_date.replace(tzinfo=None)
+                return manifest.daily_archive_start_date.date()
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/masu/database/report_manifest_db_accessor.py]: Done.

**Comment by maskarb** [comment]
@sourcery-ai review

**Review by lcouzens — DISMISSED** [review]
We have so many dates! Least this cleans some of that up. Nice job Michael.

File: koku/masu/database/report_manifest_db_accessor.py [hunk 6]
```diff
@@ -259,19 +259,18 @@ def set_manifest_daily_start_date(self, manifest_id, date):
...
         """
         Get manifest processing daily archive start date.
         Used to prevent grabbing different starts from partial processed data
         """
         manifest = self.get_manifest_by_id(manifest_id)
-        if manifest:
-            if manifest.daily_archive_start_date:
-                return manifest.daily_archive_start_date.replace(tzinfo=None)
+        if manifest and manifest.daily_archive_start_date:
+            return manifest.daily_archive_start_date.date()
```
[the remark this answers is not in the document]
Comment (maskarb) [reply koku/masu/database/report_manifest_db_accessor.py:273]: daily_archive_start_date is a Django DateTime field, so yes, it is always a datetime and will always have a date attribute.
